src/wallet/walletutil.cpp

```diff
@@ -25,3 +25,62 @@ fs::path GetWalletDir()
 
     return path;
 }
+
+static bool IsBerkeleyBtree(const fs::path& path)
+{
+    // A Berkeley DB Btree file has at least 4K.
+    // This check also prevents opening lock files.
+    boost::system::error_code ec;
+    if (fs::file_size(path, ec) < 4096) return false;
+
+    fs::ifstream file(path.string(), std::ios::binary);
+    if (!file.is_open()) return false;
+
+    file.seekg(12, std::ios::beg); // Magic bytes start at offset 12
+    uint32_t data = 0;
+    file.read((char*) &data, sizeof(data)); // Read 4 bytes of file to compare against magic
+
+    // Berkeley DB Btree magic bytes, from:
+    //  https://github.com/file/file/blob/5824af38469ec1ca9ac3ffd251e7afe9dc11e227/magic/Magdir/database#L74-L75
+    //  - big endian systems - 00 05 31 62
+    //  - little endian systems - 62 31 05 00
+    return data == 0x00053162 || data == 0x62310500;
+}
+
+std::vector<fs::path> ListWalletDir()
```

Comment:
In commit "wallet: Add WalletLocation utility class" (b35a71827bddf534816f1c5fda83c8e33a18eb63)

This does not appear to have been cherry-picked from 01a4c095c87500650663341533f000c6b613e9da correctly. The original commit is not adding any `ListWalletDir` function, so it is unclear where this is coming from. Also the function is being added without being called in this or later commits.

Reply:
Fixed.